spec/models/dialog_spec.rb

```diff
@@ -1,41 +1,58 @@
+      described_class.with_constants(:DIALOG_DIR_CORE => test_file_path) do
         expect(dialog_import_service).to receive(:import_all_service_dialogs_from_yaml_file).with(
-          test_file_path.join("seed_test.yaml").to_path)
+          Rails.root.join(test_file_path, "seed_test.yaml").to_path
+        )
         expect(dialog_import_service).to receive(:import_all_service_dialogs_from_yaml_file).with(
-          test_file_path.join("seed_test.yml").to_path)
-        Dialog.seed
+          Rails.root.join(test_file_path, "seed_test.yml").to_path
+        )
+        described_class.seed
       end
     end
 
     it "delegates to the dialog import service with a file in a sub directory" do
-      Dialog.with_constants(:DIALOG_DIR => test_file_path, :ALL_YAML_FILES => all_yaml_files) do
+      described_class.with_constants(:DIALOG_DIR_CORE => test_file_path) do
         expect(dialog_import_service).to receive(:import_all_service_dialogs_from_yaml_file).with(
-          test_file_path.join("service_dialogs/service_seed_test.yaml").to_path)
+          Rails.root.join(test_file_path, "service_dialogs/service_seed_test.yaml").to_path
+        )
         expect(dialog_import_service).to receive(:import_all_service_dialogs_from_yaml_file).with(
-          test_file_path.join("service_dialogs/service_seed_test.yml").to_path)
-        Dialog.seed
+          Rails.root.join(test_file_path, "service_dialogs/service_seed_test.yml").to_path
+        )
+        described_class.seed
       end
     end
 
     it "delegates to the dialog import service with a symlinked file" do
-      Dialog.with_constants(:DIALOG_DIR => test_file_path, :ALL_YAML_FILES => all_yaml_files) do
+      described_class.with_constants(:DIALOG_DIR_CORE => test_file_path) do
         expect(dialog_import_service).to receive(:import_all_service_dialogs_from_yaml_file).with(
-          test_file_path.join("service_dialog_symlink/service_seed_test.yaml").to_path)
+          Rails.root.join(test_file_path, "service_dialog_symlink/service_seed_test.yaml").to_path
+        )
         expect(dialog_import_service).to receive(:import_all_service_dialogs_from_yaml_file).with(
-          test_file_path.join("service_dialog_symlink/service_seed_test.yml").to_path)
-        Dialog.seed
+          Rails.root.join(test_file_path, "service_dialog_symlink/service_seed_test.yml").to_path
+        )
+        described_class.seed
+      end
+    end
+
+    it "seed files from plugins" do
+      mock_engine = double(:root => Rails.root)
+      expect(Vmdb::Plugins.instance).to receive(:registered_provider_plugins).and_return([mock_engine])
+
+      described_class.with_constants(:DIALOG_DIR_PLUGIN => test_file_path) do
+        expect(dialog_import_service).to receive(:import_all_service_dialogs_from_yaml_file).with(
+          Rails.root.join(test_file_path, "seed_test.yaml").to_path
+        )
+        described_class.seed
```

Comment:
I think this should also `expect(Dir).to receive(:glob).with(the path for the engine dialogs)`.  The `expect` above it true, but it's not related to the plugin part of the code.